nodes/object/object_visibility_output.py

```diff
@@ -0,0 +1,160 @@
+import bpy
+from bpy.props import *
+from ... events import executionCodeChanged
+from ... base_types.node import AnimationNode
+
+class ObjectVisibilityOutputNode(bpy.types.Node, AnimationNode):
+    bl_idname = "an_ObjectVisibilityOutputNode"
+    bl_label = "Object Visibility Output"
+    
+    def checkedPropertiesChanged(self, context):
+        self.updateSocketVisibility()
+        executionCodeChanged()
+    
+    #these are common obj props, the rest are type dependent
+    useView = BoolProperty(name = "Use Hide", default = False, description = "Set Hide (restrict viewport visibility)", update = checkedPropertiesChanged)
+    useSelect = BoolProperty(name = "Use Hide Select", default = False, description = "Set Hide Select (restrict viewport selection)", update = checkedPropertiesChanged)
+    useRender = BoolProperty(name = "Use Hide Render", default = False, description = "Set Hide Render (restrict rendering)", update = checkedPropertiesChanged)
+
+    useName = BoolProperty(name = "Use Name", default = False, description = "Set Show Name (visibility for object name)", update = checkedPropertiesChanged)
+    useAxis = BoolProperty(name = "Use Axis", default = False, description = "Set Show Axis (visibility for object origin and axes)", update = checkedPropertiesChanged)
+    useXray = BoolProperty(name = "Use Xray", default = False, description = "Set Show X-Ray (visibility for the object in front of others)", update = checkedPropertiesChanged)
+
+    def create(self):
+        self.inputs.new("an_ObjectSocket", "Object", "object").defaultDrawType = "PROPERTY_ONLY"
+        self.inputs.new("an_BooleanSocket", "Hide", "hide")
+        self.inputs.new("an_BooleanSocket", "Hide Select", "hideSelect")
+        self.inputs.new("an_BooleanSocket", "Hide Render", "hideRender")
+        self.inputs.new("an_BooleanSocket", "Show Name", "showName")
+        self.inputs.new("an_BooleanSocket", "Show Axis", "showAxis")
+        self.inputs.new("an_BooleanSocket", "Show Xray", "showXray")
+        self.outputs.new("an_ObjectSocket", "Object", "object")
+        for socket in self.inputs: socket.value = False
```

Comment:
should be *self.inputs[1:]* because the object socket has no value that you can set to False